% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={BHLnames},
  pdfauthor={John Doe, Jane Doe},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=2cm]{geometry}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\hypersetup{colorlinks=false, allbordercolors={0.6 0.6 1}, pdfborderstyle={/S/U/W 0.5}}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{BHLnames}
\author{John Doe, Jane Doe}
\date{2021-10-27}

\begin{document}
\maketitle

\hypertarget{abstract}{%
\section{Abstract}\label{abstract}}

\hypertarget{background}{%
\subsection{Background}\label{background}}

\begin{itemize}
\tightlist
\item
  BHL exists
\item
  CoL exists
\item
  Global Names exists
\item
  BHL has index
\item
  COL has names with nomenclatural publications
\end{itemize}

\hypertarget{new-information}{%
\subsection{New information}\label{new-information}}

\begin{itemize}
\tightlist
\item
  BHlnames now exists
\item
  BHLnames solves these 3 problems
\item
  Can access via web
\item
  Can access via API
\item
  Can see name-\textgreater description via file
\item
  Local use is possible
\end{itemize}

\hypertarget{keywords}{%
\section{Keywords}\label{keywords}}

biodiversity informatics, biodiversity heritage library, catalogue of
life, global names, nomenclature, taxonomy, library informatics

\hypertarget{classifications}{%
\section{Classifications}\label{classifications}}

\hypertarget{taxon-classification}{%
\subsection{Taxon Classification}\label{taxon-classification}}

? Animalia, Plantae, all other from the list

\hypertarget{subject-classification}{%
\subsection{Subject classification}\label{subject-classification}}

Bioinformatics

\hypertarget{funder}{%
\section{Funder}\label{funder}}

Species File Group, BHL index

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

\begin{itemize}
\item
  \begin{description}
  \tightlist
  \item[Biodiversity ``layers'']
  100mil name-strings (lexial layer)

  7mil names (nomenclatural layer)

  2.5mil taxons
  \end{description}
\end{itemize}

The Catalogue of Life (CoL) project was established in 2000 with the
goal of developing a global species checklist of all currently accepted
scientific names (Bisby 2000, Wilson and Bisby 2007). As of this
writing, CoL contains over 2 million accepted species. CoL additionally
tracks historical scientific names that are no longer accepted and
contains 1.6 million of these synonyms, which enables finding species
using either the currently accepted scientific name or their historical
synonyms. Of the 3.7 million scientific names in CoL, 53\% have
nomenclatural references to the original species description in the
literature.

Biodiversity Heritage Library (BHL) is the largest open access
collection of biodiversity literature. BHL was founded in 2005 to
increase open access to biodiversity literature and is goverened by an
internetional consortium (Gwinn and Rinaldo 2009, Rinaldo and Norton
2009, Smith and Rinaldo 2016, Kalfatovic et al. 2019). BHL currently
comprises of 270 thousand volumes and 60 million pages of biodiversity
texts. Perhaps the most common way to discover biodiversity information
in BHL is by searching for scientific names of biological organisms.

The search index of scientific names for BHL is created by Global Names
Architecture (GNA) project (Costantino 2020, Richard 2020). GNA was
founded in 2007 with the aim of providing services and tools that
globally connect biodiversity information and research via scientific
names. GNA includes nomenclatural (ZooBank, a zoological names
registrator) and lexical tools (GNfinder, GNverifier, GNparser). The
GNfinder tool can be used to detect potential scientific names within
text documents by searching for text patterns that follow the Linnaean
nomenclature system (Linnaeus 1753). GNverifier enables searching for
scientific names in more than 100 sources of biodiversity data, which is
used to help verify if potential scientific names found by GNfinder in
text documents are actually scientific names. GNverifier uses data from
Catalogue of Life, Encyclopedia of Life, Index Fungorum, Global
Biodiversity Information Facility and many others. GNparser normalizes
scientific names and extracts the semantic meanings of each component
(e.g., interprets the scientific name ``Tyrannosaurus rex Osborn, 1905''
into genus Tyrannosaurus, species epithet rex, author Osborn, and year
of publication 1905). Normalizing scientific names also enables
organizing lexical variants of a name into a lexical group.

The BHL search index of scientific names is created by processing BHL
data through another GNA tool, BHLindex. BHLindex breaks each volume
into tokenized words, which enables the heuristic and natural language
processing algorithms of GNfinder to detect potential scientific names
within the volume's text. Potential scientific names are verified
(reconciled and resolved) via GNverifier. The process of generating the
BHL scientific names index has been highly optimized for performance,
which enables processing 60 million BHL pages in \textasciitilde12 hours
on a modern laptop. The speed of the BHL index creation offers the major
advantage of being able to continuously improve BHL search as the
scanned page OCR improves (Herrmann 2020), and to incrementally improve
the scientiifc name finding algorithms of GNA tools. GNverifier also
augments the metadata for each BHL volume by attempting to determine the
kingdom and lowest clade that contain the most scientific names within
the BHL volume using management classification from Catalogue of Life
(Gordon 2009).

Searching BHL for a biological organism using only its currently
accepted scientific name could exclude a significant amount of
information about the organism due to changes in taxon names over time.
CoL currently contains \textasciitilde3.6 million scientific names for
\textasciitilde2 million taxa. The database of GNverifier contains 8
million scientific names, for \textasciitilde2.5 million known taxa.
From this data we can estimate that there are 2-3 names per taxon on
average, which fits with other published estimates (3.5 names/taxon in
seed plants (Scotland and Wortley 2003), but synonymy can vary by group
with only 1.2 names/taxon in stoneflies (DeWalt and Ower 2019)). To
obtain more comprehensive knowledge about a biological organism, it is
important to search not by just one scientific name, but by all synonyms
used for the organism historically.

Another way to access information in BHL is by searching for a
particular paper. Thanks to the BioStor project, metadata and page
locations are known for 312,126 papers in BHL (Page 2010). However,
designing software to automatically find any paper in BHL by a citation
string is still challenging. There are a large variety of lexical
variants for every journal name (FIG). No standard for citation
formatting was ever agreed upon by publishers, so the arrangement of
authors, year of publication, volume, issue, and page widely varies. It
is much easier to search for a year, volume and page data as numbers,
but in BHL numeric metadata is stored as a string and is not
consistently formatted throughout the corpus with many journals in the
past using Roman numerals. Quite often metadata is missing entirely or
contains errors, which complicates automatically linking citation
strings to the correct page in BHL.

TODO: incorporate (Pilsk et al. 2010) which covers librarian point of
view on metadata Pilsk, et al.: ``Over the centuries, scientific
nomenclature specialists have been forming species citations using
abbreviations and notes that accommodate their spe- cial fields of
study. These practices developed in an isolated manner specific to
discipline (or subdiscipline) and independent of related species
projects or the expertise of librarians and information professionals.
Concurrently, librarians developing and implementing metadata policies
and procedures never entered into conversation with taxonomists.
Librarians did little to dis- cover how taxonomic citations are formed
and what the actual access needs of the scientists were. As a result,
each group adopted its own way of pro- cessing information, and constant
translation between the two worlds was necessary. At best, a clumsy, but
acceptable, disconnected reliance existed between these two fields''

TODO: Fig: examples of journal name different spellings

Even if a researcher has the citation for a nomenclatural event, it is
often not a trivial task to navigate to the exact page in BHL in order
to access the original species description for research. Citations are
hard to match to pages in BHL as described above. Sometimes OCR errors
change a name spelling and as a result exist in a volume under different
spelling variants.

Quite often people add an annotation next to new species or combination,
such as sp.nov, comb nov. However such annotations are not consistent
(FIG), and also often modified by OCR. When several species of the same
genus are described, the genus in the name is often abbreviated, which
also masks the name from a researcher.

TODO: Fig: examples for annotations

The project we describe in this paper helps to adress these problems
with the following solutions: 1) BHLnames uses a combination of the
names index, CoL data, and BHL metadata to improve nomenclatural and
taxonomic search, link a citation string to a page in BHL, or link to
pages containing the original species description. 2) BHLnames
normalizes BHL data for publication years and pages, making it easier to
compare numeric representations. 3) The project uses 2,085,819 million
citations of nomenclatural events for names in CoL to provide data about
nomenclatural events. 4) BHLnames normalizes journal names in order to
match citation strings to the journal name in BHL. 5) We use a Naive
Bayes algorithm to calculate probabilities of a citation to BHL page
match and return the best results. The data linking CoL nomenclatural
citations to the original species description pages in BHL is
downloadable from the Dryad data repository (Link).

\hypertarget{project-description}{%
\section{Project description}\label{project-description}}

Title: BHLnames

Study area description: Biodiversity Informatics

Design description: This is the whole paper actually (?)

\hypertarget{goals}{%
\subsection{Goals}\label{goals}}

The goals of BHLnames project are to:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Dramatically simplify citation string search using the names index.

  Use citation metadata in combination with occurences of a scientific
  name to link to pages that potentially match the citation with a
  provided probability score.
\item
  Link to original species descriptions and nomenclatural events in BHL.

  Use citation search in combination with CoL nomenclatural citations to
  build an nomenclatural event index that will increase usability of BHL
  for biodiversity researchers.
\item
  Add ``taxonomic intalligence'' to the BHL's index of scientific names.

  Supplement the nomenclatural event index with CoL synonym data to
  optionally extend BHL name search to include all relevent taxonomical
  information by returning search results for the accepted name and all
  of its synonyms.
\end{enumerate}

(Methods)

\hypertarget{technical-introduction}{%
\subsection{Technical introduction}\label{technical-introduction}}

BHLnames an Open Source project written in Go and JavaScript. It can be
freely used by anyone with Internet connection, write access to a
PostgreSQL database, and XXX GB of disk space. It consists of one
stand-alone binary file and requires no additional dynamic libraries.
The file can be downloaded at:
https://github.com/gnames/bhlnames/releases Useage requires an
initialization stage that can take several hours to complete, but
afterwards the BHLnames binary can be used without an internet
connection.

\hypertarget{architecture}{%
\subsection{Architecture}\label{architecture}}

BHLnames' code structure mostly adheres to Clean Architecture principles
(Martin 2018). The main use-case functionality is defined in an
interface (Link?). The core (entity) modules that do not require
input/output (I/O) interactions with a database, the internet or local
storage, are located in the `ent' directory. The `ent' directory also
contains interfaces implemented by modules that need I/O access. The
Clean Architecture approach is advantageous because it uncouples core
modules from particular implementations of external I/O processes and
provides flexibility for changing ways of storing and accessing the
data. Modules requiring database, internet, user interfaces, or
framework accesss implement entity interfaces and depend on the main
use-case interface for providing functionality of BHLnames. One
exception to this rule is the `bhlnames' directory that contains the
commandline interface, following convention established by its Cobra
module dependency.

BHLnames provides 3 user interfaces:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  The commandline application starts and stops the web service or
  provides BHLnames functionality via the terminal.
\item
  The RESTful API provides endpoints for remote access to BHLnames
  functionality via HTTP calls.
\item
  The web interface is a Javascript widget that can be embedded on any
  website to connect scientific names with BHL literature in an
  interactive pop-up modal.
\end{enumerate}

\hypertarget{dependencies}{%
\subsubsection{Dependencies}\label{dependencies}}

Initialization phase of the project requires internet access to a data
export from BHL (https://www.biodiversitylibrary.org/data/data.zip) and
the BHLindex RESTful API.

BHLindex is an Open Source application written in Go that creates a
scientific names index of BHL. The application scans BHL's 60 million
pages, detects scientific names and provides data about their page
location via RESTful interface. It is possible to use publicly available
URL that serves BHLindex API at (LINK).

There are many dependencies on Go modules, some of them written
specifically for Global Names project. We will talk about some of GN
modules further. Go dependencies are required only during the
compilation of the project and are not needed for running BHLnames
service or commandline application.

\hypertarget{workflow}{%
\subsection{Workflow}\label{workflow}}

\hypertarget{data-import}{%
\subsubsection{Data Import}\label{data-import}}

\hypertarget{interfaces}{%
\subsection{Interfaces}\label{interfaces}}

\hypertarget{gui}{%
\subsubsection{GUI}\label{gui}}

\hypertarget{embeddable-widget}{%
\subsubsection{Embeddable widget}\label{embeddable-widget}}

\hypertarget{api}{%
\subsubsection{API}\label{api}}

\hypertarget{command-line}{%
\subsubsection{Command line}\label{command-line}}

\hypertarget{input-formats}{%
\subsubsection{Input formats}\label{input-formats}}

\hypertarget{audience}{%
\subsection{Audience}\label{audience}}

CoL ZooBank WoRMS? TW BHL users Taxonomists Librarians Citizen
Scientists Nomenclators Other Reseachers

(Results)

\hypertarget{examples-of-usages}{%
\subsubsection{Examples of usages}\label{examples-of-usages}}

\hypertarget{summary-of-scores}{%
\subsubsection{Summary of scores}\label{summary-of-scores}}

\hypertarget{benchmarks}{%
\subsubsection{Benchmarks}\label{benchmarks}}

\hypertarget{open-data-deliverables}{%
\subsubsection{Open data deliverables}\label{open-data-deliverables}}

(Discussion)

\hypertarget{conclusions}{%
\subsection{Conclusions}\label{conclusions}}

\begin{itemize}
\tightlist
\item
  Jigsaw puzzle where a name is the first piece.
\item
  We add more pieces
\end{itemize}

Advantages, impact

Missing parts

Ways to improve

How it helps for the other future developments

How to contribute?

Funding: skip it here? Same as above

\hypertarget{web-location-uris}{%
\section{Web location (URIs)}\label{web-location-uris}}

Homepage: bhlnames.globalnames.org

Wiki: github.com/gnames/bhlnames wiki (maybe, or nothing)

Download page: data or code? * code: https://github.com/gnames/bhlnames
* data: some place on Dryad?

Download mirror: n/a

Bug database: https://github.com/gnames/bhlnames/issues

Mailing list: https://gitter.io/gnames/something ?

Blog: http://globalnames.org

Vendor: N/A ask Matt

\hypertarget{technical-specification}{%
\section{Technical specification}\label{technical-specification}}

Platform: ?

Programming language: Go, JavaScript?

Operating system: GNU/Linux

Interface language: JavaScript?

Service endpoint: * https://bionames.globalnames.org/api/v1 *
https://apidoc.globalnames.org/bhlnames

\hypertarget{repository}{%
\section{Repository}\label{repository}}

Type: Git

Browse URI: https://github.com/gnames/bhlnames

Location: ?

Module: ?

Anonymous root: n/a

\hypertarget{usage-license}{%
\section{Usage license}\label{usage-license}}

MIT

\hypertarget{implementation}{%
\section{Implementation}\label{implementation}}

Implements specification: N/A

Audience:

\hypertarget{additional-information}{%
\section{Additional information}\label{additional-information}}

\hypertarget{acknowledgements}{%
\section{Acknowledgements}\label{acknowledgements}}

Thanks to you all. Rich Pyle for journal data

\hypertarget{author-contributions}{%
\section{Author contributions}\label{author-contributions}}

\hypertarget{references}{%
\section{References}\label{references}}

\hypertarget{figures}{%
\section{Figures}\label{figures}}

\hypertarget{tables}{%
\section{Tables}\label{tables}}

\hypertarget{supplementary-material}{%
\section{Supplementary material}\label{supplementary-material}}

\hypertarget{endnotes}{%
\section*{Endnotes}\label{endnotes}}
\addcontentsline{toc}{section}{Endnotes}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Bisby2000}{}}%
Bisby FA (2000) {The quiet revolution: Biodiversity informatics and the
Internet}. Science 289: 2309--2312.
\url{https://doi.org/10.1126/science.289.5488.2309}

\leavevmode\vadjust pre{\hypertarget{ref-Costantino2020}{}}%
Costantino G (2020) {BHL Improves the Speed and Accuracy of its
Taxonomic Name Finding Services with gnfinder}. Available from:
\url{https://blog.biodiversitylibrary.org/2020/07/bhl-improvestaxonomic-name-services-gnfinder.html}.

\leavevmode\vadjust pre{\hypertarget{ref-DeWalt2019}{}}%
DeWalt RE, Ower GD (2019) Ecosystem services, global diversity, and rate
of stonefly species descriptions (insecta: plecoptera). 10.
\url{https://doi.org/10.3390/insects10040099}

\leavevmode\vadjust pre{\hypertarget{ref-Gordon2009}{}}%
Gordon DP (2009) {Towards a management hierarchy (classification) for
the Catalogue of Life: Draft Discussion Document}. Available from:
\url{https://web.archive.org/web/20090330224421/http://www.catalogueoflife.org/annual-checklist/2009/info\%7B/_\%7Dhierarchy.php}.

\leavevmode\vadjust pre{\hypertarget{ref-Gwinn2009}{}}%
Gwinn NE, Rinaldo C (2009) {The Biodiversity Heritage Library: Sharing
biodiversity literature with the world}. IFLA Journal 35: 25--34.
\url{https://doi.org/10.1177/0340035208102032}

\leavevmode\vadjust pre{\hypertarget{ref-Herrmann2020}{}}%
Herrmann E (2020) {Building the Biodiversity Heritage Library's
Technical Strategy}. Biodiversity Information Science and Standards 4.
\url{https://doi.org/10.3897/biss.4.59084}

\leavevmode\vadjust pre{\hypertarget{ref-Kalfatovic2019}{}}%
Kalfatovic MR, Costantino G, Rinaldo CA (2019) {The Biodiversity
Heritage Library: Unveiling a World of Knowledge About Life on Earth}.
In:, 352--355. \url{https://doi.org/10.1007/978-3-030-30760-8_32}

\leavevmode\vadjust pre{\hypertarget{ref-Linnaeus1753}{}}%
Linnaeus C (1753) vol. 1 Species plantarum: Exhibentes plantas rite
cognitas, ad genera relatas, cum differentiis specificis, nominibus
trivialibus, synonymis selectis, locis natalibus, secundum systema
sexuale digestas. Stockholm: Impensis Laurentii Salvii.
\url{https://doi.org/10.5962/bhl.title.669}

\leavevmode\vadjust pre{\hypertarget{ref-Martin2018}{}}%
Martin RC (2018) Clean architecture: A craftsman's guide to software
structure and design. Prentice Hall. Available from:
\url{https://books.google.com/books?id=8ngAkAEACAAJ}.

\leavevmode\vadjust pre{\hypertarget{ref-Page2010}{}}%
Page R (2010) {Finding scientific articles in a large digital archive:
BioStor and the Biodiversity Heritage Library}. Nature Precedings:
1--19. \url{https://doi.org/10.1038/npre.2010.4928.1}

\leavevmode\vadjust pre{\hypertarget{ref-Pilsk2010}{}}%
Pilsk SC, Person MA, Deveer JM, Furfey JF, Kalfatovic MR (2010) {The
Biodiversity Heritage Library: Advancing Metadata Practices in a
Collaborative Digital Library}. Journal of Library Metadata 10:
136--155. \url{https://doi.org/10.1080/19386389.2010.506400}

\leavevmode\vadjust pre{\hypertarget{ref-Richard2020}{}}%
Richard J (2020) {Improving Taxonomic Name Finding in the Biodiversity
Heritage Library}. Biodiversity Information Science and Standards 4.
\url{https://doi.org/10.3897/biss.4.58482}

\leavevmode\vadjust pre{\hypertarget{ref-Rinaldo2009}{}}%
Rinaldo C, Norton C (2009) {BHL, The Biodiversity Heritage Library: An
Expanding International Collaboration}. Nature Precedings.
\url{https://doi.org/10.1038/npre.2009.3620.1}

\leavevmode\vadjust pre{\hypertarget{ref-Scotland2003}{}}%
Scotland RW, Wortley AH (2003) {How many species of seed plants are
there?} Taxon 52: 101--104.

\leavevmode\vadjust pre{\hypertarget{ref-Smith2016}{}}%
Smith JE, Rinaldo CA (2016) {Collaborating on open science: The journey
of the Biodiversity Heritage Library}. Dobreva M, Schmidt B (Eds).
Information Services \& Use 35: 211--216.
\url{https://doi.org/10.3233/ISU-150785}

\leavevmode\vadjust pre{\hypertarget{ref-Wilson2007}{}}%
Wilson K, Bisby F (2007) {The Catalogue of Life: indexing the world's
species}. BGjournal 4: 24--29.

\end{CSLReferences}

\end{document}
