% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={BHLnames},
  pdfauthor={John Doe, Jane Doe},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=2cm]{geometry}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\hypersetup{colorlinks=false, allbordercolors={0.6 0.6 1}, pdfborderstyle={/S/U/W 0.5}}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{BHLnames}
\author{John Doe, Jane Doe}
\date{2021-10-27}

\begin{document}
\maketitle

\hypertarget{abstract}{%
\section{Abstract}\label{abstract}}

\hypertarget{background}{%
\subsection{Background}\label{background}}

\begin{itemize}
\tightlist
\item
  BHL exists
\item
  CoL exists
\item
  Global Names exists
\item
  BHL has index
\item
  COL has names with nomenclatural publications
\end{itemize}

\hypertarget{new-information}{%
\subsection{New information}\label{new-information}}

\begin{itemize}
\tightlist
\item
  BHlnames now exists
\item
  BHLnames solves these 3 problems
\item
  Can access via web
\item
  Can access via API
\item
  Can see name-\textgreater description via file
\item
  Local use is possible
\end{itemize}

\hypertarget{keywords}{%
\section{Keywords}\label{keywords}}

biodiversity informatics, biodiversity heritage library, catalogue of
life, global names, nomenclature, taxonomy, library informatics

\hypertarget{classifications}{%
\section{Classifications}\label{classifications}}

\hypertarget{taxon-classification}{%
\subsection{Taxon Classification}\label{taxon-classification}}

? Animalia, Plantae, all other from the list

\hypertarget{subject-classification}{%
\subsection{Subject classification}\label{subject-classification}}

Bioinformatics

\hypertarget{funder}{%
\section{Funder}\label{funder}}

Species File Group, BHL index

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

\begin{itemize}
\item
  \begin{description}
  \tightlist
  \item[Biodiversity ``layers'']
  100mil name-strings (lexial layer)

  7mil names (nomenclatural layer)

  2.5mil taxons
  \end{description}
\item
  CoL

  \begin{itemize}
  \tightlist
  \item
    exists since ?
  \item
    goal all species (global scope)
  \item
    covers taxonomy and, to a degree, nomenclatural layers
  \item
    has \textasciitilde4mil name-strings, \textasciitilde2m taxons
  \item
    about ?\% of name-strings have references
  \item
    taxon/synonymy info allows to find connection from name to taxon to
    synonyms
  \end{itemize}
\end{itemize}

Biodiversity Heritage Library (BHL) is the largest open access
collection of biodiversity literature. BHL was founded in 2006 and is
goverened by an internetional consortium. BHL currently comprises of 270
thousand volumes and 60 million pages of biodiversity texts. By far the
best way to discover biodiversity information in BHL is by searching for
scintific names of organisms. The index of scientific names for BHL is
created by Global Names Architecture (GNA) project.

The purpose of GNA is creation of services and tools that allow to
globally connect biodiversity information and research via scientific
names. GNA was founded in 2007 and includes nomenclatural (ZooBank, a
zoological names registrator) and lexical tools (GNparser, GNverifier,
GNfinder). GNfinder detects scientific names in texts. GNparser
normalizes scientific name-strings and extracts semantic meanings of
their components. Normalised scientific names allow to aggregate lexical
variants of a name into a lexical group. GNverifier allows to search a
name-string of interest in more than 100 sources of biodiversity data.
GNverifier uses data from Catalogue of Life, Encyclopedia of Life, Index
Fungorum, Global Biodiversity Information Facility and many others. GNA
tools are used for creation of scientific name index in BHL.

The BHL index of scientific names is created by processing BHL data
through BHLindex, yet another GNA tool. BHLindex breaks each volume in
BHL into tokens (words). Then it scans the tokens and uses heuristic and
natural language processing algorithms of GNfinder to detect possible
scientific names. Later these names are verified (reconciled and
resolved) via GNverifier. The process is quite fast and allows to
process 60 millions of BHL pages in about 12 hours on a modern laptop.
The efficiency of the index creation allows to incrementally improve
algorithms and quality of the name-finding. GNverifier also provides
metadata for each volume, calculating its biological ``context''. Such
information includes the kingdom which contains most of the names found
in a volume, as well as the lowest clade that contains more than half of
the volume' names. The calculation uses managerial classification of the
Catalogue of Life.

\begin{itemize}
\tightlist
\item
  \textasciitilde3 names per taxon, examples: like ``Puma concolor'',
  ``Felis concolor'' etc

  \begin{itemize}
  \tightlist
  \item
    finding all information about species is hard, because of changing
    names
  \end{itemize}
\item
  Hard to find a paper in BHL by having a citation (especially
  automatically)

  \begin{itemize}
  \tightlist
  \item
    not enough meta information
  \item
    metadata problems: year, page, volume information are ``strings'',
    meaning they need to be parsed to compare with numbers
  \item
    sometimes metadata is misleading (example?) missing
  \item
    citations parsing is hard
  \item
    inconsistent journal abbreviation and naming in different sources
  \end{itemize}
\item
  Hard to find nomenclatural events from name and citation

  \begin{itemize}
  \tightlist
  \item
    citation look above
  \item
    name: OCR errors can change name,
  \item
    name might exist but on a different page (like book index etc\ldots)
  \item
    name: annotations (sp.nov., comb. nov etc are not consistent)
  \item
    name: abbreviated name problem (not solved yet)
  \end{itemize}
\item
  Solution to these problems

  \begin{itemize}
  \tightlist
  \item
    use COL data that connects name-strin to ref.
  \item
    use CoL accepted/synonym information to get connect names-string to
    taxon
  \item
    use name index in BHL together with COL data to solve these 3
    problems.
  \end{itemize}
\item
  BHLnames project tries to solve these problems using aforementioned
  stuff
\end{itemize}

\hypertarget{project-description}{%
\section{Project description}\label{project-description}}

Title: BHLnames

Study area description: Biodiversity Informatics

Design description: This is the whole paper actually (?)

\hypertarget{goals}{%
\subsection{Goals}\label{goals}}

(Methods)

\hypertarget{architecuture}{%
\subsection{Architecuture}\label{architecuture}}

\hypertarget{dependencies}{%
\subsubsection{Dependencies}\label{dependencies}}

\hypertarget{project-structure}{%
\subsubsection{Project structure}\label{project-structure}}

\hypertarget{workflow}{%
\subsubsection{Workflow}\label{workflow}}

\hypertarget{data-import}{%
\paragraph{Data Import}\label{data-import}}

\hypertarget{interfaces}{%
\subsection{Interfaces}\label{interfaces}}

\hypertarget{gui}{%
\subsubsection{GUI}\label{gui}}

\hypertarget{embeddable-widget}{%
\subsubsection{Embeddable widget}\label{embeddable-widget}}

\hypertarget{api}{%
\subsubsection{API}\label{api}}

\hypertarget{command-line}{%
\subsubsection{Command line}\label{command-line}}

\hypertarget{input-formats}{%
\subsubsection{Input formats}\label{input-formats}}

\hypertarget{audience}{%
\subsection{Audience}\label{audience}}

CoL ZooBank WoRMS? TW BHL users Taxonomists Librarians Citizen
Scientists Nomenclators Other Reseachers

(Results)

\hypertarget{examples-of-usages}{%
\subsubsection{Examples of usages}\label{examples-of-usages}}

\hypertarget{summary-of-scores}{%
\subsubsection{Summary of scores}\label{summary-of-scores}}

\hypertarget{benchmarks}{%
\subsubsection{Benchmarks}\label{benchmarks}}

\hypertarget{open-data-deliverables}{%
\subsubsection{Open data deliverables}\label{open-data-deliverables}}

(Discussion)

\hypertarget{conclusions}{%
\subsection{Conclusions}\label{conclusions}}

Advantages, impact

Missing parts

Ways to improve

How it helps for the other future developments

How to contribute?

Funding: skip it here? Same as above

\hypertarget{web-location-uris}{%
\section{Web location (URIs)}\label{web-location-uris}}

Homepage: bhlnames.globalnames.org

Wiki: github.com/gnames/bhlnames wiki (maybe, or nothing)

Download page: data or code? * code: https://github.com/gnames/bhlnames
* data: some place on Dryad?

Download mirror: n/a

Bug database: https://github.com/gnames/bhlnames/issues

Mailing list: https://gitter.io/gnames/something ?

Blog: http://globalnames.org

Vendor: N/A ask Matt

\hypertarget{technical-specification}{%
\section{Technical specification}\label{technical-specification}}

Platform: ?

Programming language: Go, JavaScript?

Operating system: GNU/Linux

Interface language: JavaScript?

Service endpoint: * https://bionames.globalnames.org/api/v1 *
https://apidoc.globalnames.org/bhlnames

\hypertarget{repository}{%
\section{Repository}\label{repository}}

Type: Git

Browse URI: https://github.com/gnames/bhlnames

Location: ?

Module: ?

Anonymous root: n/a

\hypertarget{usage-license}{%
\section{Usage license}\label{usage-license}}

MIT

\hypertarget{implementation}{%
\section{Implementation}\label{implementation}}

Implements specification: N/A

Audience:

\hypertarget{additional-information}{%
\section{Additional information}\label{additional-information}}

\hypertarget{acknowledgements}{%
\section{Acknowledgements}\label{acknowledgements}}

Thanks to you all. Rich Pyle for journal data

\hypertarget{author-contributions}{%
\section{Author contributions}\label{author-contributions}}

\hypertarget{references}{%
\section{References}\label{references}}

\hypertarget{figures}{%
\section{Figures}\label{figures}}

\hypertarget{tables}{%
\section{Tables}\label{tables}}

\hypertarget{supplementary-material}{%
\section{Supplementary material}\label{supplementary-material}}

\hypertarget{endnotes}{%
\section{Endnotes}\label{endnotes}}

\end{document}
