% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={BHLnames},
  pdfauthor={John Doe, Jane Doe},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=2cm]{geometry}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\hypersetup{colorlinks=false, allbordercolors={0.6 0.6 1}, pdfborderstyle={/S/U/W 0.5}}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{BHLnames}
\author{John Doe, Jane Doe}
\date{2021-10-27}

\begin{document}
\maketitle

\hypertarget{abstract}{%
\section{Abstract}\label{abstract}}

\hypertarget{background}{%
\subsection{Background}\label{background}}

\begin{itemize}
\tightlist
\item
  BHL exists
\item
  CoL exists
\item
  Global Names exists
\item
  BHL has index
\item
  COL has names with nomenclatural publications
\end{itemize}

\hypertarget{new-information}{%
\subsection{New information}\label{new-information}}

\begin{itemize}
\tightlist
\item
  BHlnames now exists
\item
  BHLnames solves these 3 problems
\item
  Can access via web
\item
  Can access via API
\item
  Can see name-\textgreater description via file
\item
  Local use is possible
\end{itemize}

\hypertarget{keywords}{%
\section{Keywords}\label{keywords}}

biodiversity informatics, biodiversity heritage library, catalogue of
life, global names, nomenclature, taxonomy, library informatics

\hypertarget{classifications}{%
\section{Classifications}\label{classifications}}

\hypertarget{taxon-classification}{%
\subsection{Taxon Classification}\label{taxon-classification}}

? Animalia, Plantae, all other from the list

\hypertarget{subject-classification}{%
\subsection{Subject classification}\label{subject-classification}}

Bioinformatics

\hypertarget{funder}{%
\section{Funder}\label{funder}}

Species File Group, BHL index

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

\begin{itemize}
\item
  \begin{description}
  \tightlist
  \item[Biodiversity ``layers'']
  100mil name-strings (lexial layer)

  7mil names (nomenclatural layer)

  2.5mil taxons
  \end{description}
\end{itemize}

The Catalogue of Life (CoL) project was established in 2000 with the
goal of developing a global species checklist of all currently accepted
scientific names (Bisby 2000). As of this writing, CoL contains over 2
million accepted species. CoL additionally tracks historical scientific
names that are no longer accepted and contains 1.6 million of these
synonyms, which enables finding species using either the currently
accepted scientific name or their historical names. Of the 3.7 million
scientific names in CoL, 53\% have nomenclatural references to the
original species description in the literature.

Biodiversity Heritage Library (BHL) is the largest open access
collection of biodiversity literature. BHL was founded in 2005 to
increase open access to biodiversity literature and is goverened by an
internetional consortium (Gwinn and Rinaldo 2009, Rinaldo and Norton
2009, Smith and Rinaldo 2016, Kalfatovic et al. 2019). BHL currently
comprises of 270 thousand volumes and 60 million pages of biodiversity
texts. By far the best way to discover biodiversity information in BHL
is by searching for scintific names of organisms. The index of
scientific names for BHL is created by Global Names Architecture (GNA)
project (Costantino 2020, Richard 2020).

The purpose of GNA is creation of services and tools that globally
connect biodiversity information and research via scientific names. GNA
was founded in 2007 and includes nomenclatural (ZooBank, a zoological
names registrator) and lexical tools (GNfinder, GNparser, GNverifier).
GNfinder detects scientific names in texts. GNparser normalizes
scientific name-strings and extracts semantic meanings of their
components. Normalizing scientific names also enables aggregating
lexical variants of a name into a lexical group. GNverifier enables
searching for scientific names in more than 100 sources of biodiversity
data, which helps verify whether the searched names string is a
scientific name. GNverifier uses data from Catalogue of Life,
Encyclopedia of Life, Index Fungorum, Global Biodiversity Information
Facility and many others. These GNA tools are used for creating the
scientific name index in BHL, which is used for BHL search (Costantino
2020, Richard 2020).

The BHL index of scientific names is created by processing BHL data
through another GNA tool, BHLindex. BHLindex breaks each volume in BHL
into tokens (words). The heuristic and natural language processing
algorithms of GNfinder are used to detect possible scientific names in
the tokenized words. Later these names are verified (reconciled and
resolved) via GNverifier. The process of generating the BHL scientific
names index has been highly optimized for speed, which enables
processing 60 million BHL pages in \textasciitilde12 hours on a modern
laptop. The efficiency of the index creation offers the major advantage
of being able to improve BHL search as the scanned page OCR improves,
and to incrementally improve the scientiifc name finding algorithms of
GNA tools. GNverifier also enhances the metadata for each volume by
attempting to determine the biological ``context''. Such information
includes the kingdom which contains most of the names found in a volume,
as well as the lowest clade that contains more than half of the volume'
names. The calculation uses the management classification from Catalogue
of Life (\textbf{Wellington2009?}).

Searching BHL for a particular name in the name index would often miss a
significant amount of data because taxons' names do change with time.
CoL currently contains \textasciitilde4 million names, for
\textasciitilde2 million taxons. The database of GNverifier contains 8
million canonical forms, for \textasciitilde2.5 million known taxons.
From this data we can estimate that there are 2-3 names per a taxon on
average. It means that to get more comprehensive data about species it
is important to search not only by one name, but by all names used for
the species historically.

Another way to aquire information from BHL is by searching for a
particular paper. Thanks to BioStore effort metainformation and location
is known for 200 000? papers in BHL (Page 2010). However to find a paper
by a citation is still hard, especially automatically by a script. There
is a large variety of lexical variants for every journal name (FIG).
Format of data about the year of publication, a journal volume, pages
differs from one citation to the next. It is much easier to search for a
year, volume and page data as numbers, but in BHL such metadata is
stored as a string and is not consistent throghout the corpus. Quite
often such metadata is missing, or contains mistakes and it makes the
search harder.

TODO: incorporate (Pilsk et al. 2010) which covers librarian point of
view on metadata Pilsk, et al.: ``Over the centuries, scientific
nomenclature specialists have been forming species citations using
abbreviations and notes that accommodate their spe- cial fields of
study. These practices developed in an isolated manner specific to
discipline (or subdiscipline) and independent of related species
projects or the expertise of librarians and information professionals.
Concurrently, librarians developing and implementing metadata policies
and procedures never entered into conversation with taxonomists.
Librarians did little to dis- cover how taxonomic citations are formed
and what the actual access needs of the scientists were. As a result,
each group adopted its own way of pro- cessing information, and constant
translation between the two worlds was necessary. At best, a clumsy, but
acceptable, disconnected reliance existed between these two fields''

TODO: Fig: examples of journal name different spellings

If a researcher has a citation of a nomenclatural event and tries to
find the paper with original description of species, it is often not a
trivial task. Citations are hard to match to pages in BHL as described
above. Sometimes OCR errors change a name spelling and as a result exist
in a volume under different spelling variants. Quite often people add an
annotation next to new species or combination, such as sp.nov, comb nov.

However such annotations are not consistent (FIG), and also often
modified by OCR. When several species of the same genus are described,
the genus in the name is often abbreviated, which also masks the name
from a researcher.

TODO: Fig: examples for annotations

The project we describe in this paper helps to adress these problems
with the following solutions. BHLnames uses name index, CoL data, and
BHL metadata together to improve search of an information connected to a
name-string, match a citation to a page in BHL, find pages that contain
original descriptions of species.. BHLnames uses CoL synonymy data to
optionally expand name search to all its synonyms. BHLnames normalizes
BHL data for publication years and pages, making it easier to compare it
with numeric representations of years and pages. The project uses 2?
million citations of nomenclatural events for names in CoL to provide
data about nomenclatural events. BHLnames normalizes journal names in a
citation and BHL to match them together. We use Naive Bayes approach to
calculate probabilities of a citation to a BHL page match and return the
best results. Such results are provided via Dryad (Link)

\hypertarget{project-description}{%
\section{Project description}\label{project-description}}

Title: BHLnames

Study area description: Biodiversity Informatics

Design description: This is the whole paper actually (?)

\hypertarget{goals}{%
\subsection{Goals}\label{goals}}

The goals of BHLnames project are:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Provide an ability to search for a citation in BHL using citation
  metadata and the index of names data.

  Use names' index information to dramatically simplify search for a
  publication. Combine citation metadata together with occurences of a
  name to point to possible pages of a citation with high probability.
\item
  Discover original descriptions and nomenclatural events in BHL.

  Use the goal \#1 together with citation and name information from CoL.
  Combine these data to determine pages containing nomenclatural events
  for scientific names. Increase usability of BHL for biodiversity
  reseearchers by augmenting scientific names index with the
  nomenclatural events index.
\item
  Add ``taxonomic intalligence'' to the BHL's index of scientific names.

  Use goal \#2 and the CoL data to optionally expand BHL name search
  with taxonomical information. Return data not only for a particular
  name, but for its synonyms. Provide data about currently accepted name
  for the taxa. Also, when available, provide data about nomenclatural
  events for all relevant names.
\end{enumerate}

(Methods)

\hypertarget{technical-introduction}{%
\subsection{Technical introduction}\label{technical-introduction}}

BHLnames an Open Source project written in Go and JavaScript. It can be
used independently by anybody with Internet connection and a write
access to a PostgreSQL database. It consists of one stand-alone file and
requires no additional dynamic libraries. The file can be downloaded at
(GitHub link)

BHLnames requires a substantial initialization stage. After
initialization it can be used in off-line mode.

\hypertarget{architecture}{%
\subsection{Architecture}\label{architecture}}

BHLnames' code structure mostly uses Clean Architecture principles for
organisation of its modules. The main use-case functinality is defined
in an in an interface (Link?).

The core (entity) modules that do not require I/O interactions with a
database, internet or local storate, are located in `ent' directory. The
`ent' directory also contains interfaces implemented by modules that
need I/O access. Such approach uncouples core modules from particular
implementations of external I/O processes and provides flexibility for
changing ways of storing and accessing the data.

The modules that require access to database, internet, interfaces to
users, accesses to frameworks implement entity interfaces and depend on
the main use-case interface for providing functionality of BHLnames. One
exception to this rule is `bhlnames' directory that contains
command-line interface to the project.

There are 3 different kinds of user-interfaces:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Command-line application starts and stops the web-service or provides
  BHLnames functionality via terminal window.
\item
  RESTful API provides endpoints for remote access to BHLnames
  functionality via HTTP calls.
\item
  Web-interface allows remote interactive access BHLnames users via
  Internet.
\end{enumerate}

\hypertarget{dependencies}{%
\subsubsection{Dependencies}\label{dependencies}}

Initialization phase of the projct requires a remote access to remote
data-dump of BHL located at (LINK). Also it needs an access to a RESTful
inteface of BHLindex application.

BHLindex is an Open Source application written in Go that creates
scientific names index of BHL. The application scans BHL's 60 million
pages, detects scientific names and provides data about their occurences
via RESTful interface. It is possible to use publicly available URL that
serves BHLindex API at (LINK).

There are many dependencies on Go modules, some of them written
specifically for Global Names project. We will talk about some of GN
modules further. Go dependencies are required only during the
compilation of the project and are not needed for running BHLnames
service or command-line application.

\hypertarget{workflow}{%
\subsection{Workflow}\label{workflow}}

\hypertarget{data-import}{%
\subsubsection{Data Import}\label{data-import}}

\hypertarget{interfaces}{%
\subsection{Interfaces}\label{interfaces}}

\hypertarget{gui}{%
\subsubsection{GUI}\label{gui}}

\hypertarget{embeddable-widget}{%
\subsubsection{Embeddable widget}\label{embeddable-widget}}

\hypertarget{api}{%
\subsubsection{API}\label{api}}

\hypertarget{command-line}{%
\subsubsection{Command line}\label{command-line}}

\hypertarget{input-formats}{%
\subsubsection{Input formats}\label{input-formats}}

\hypertarget{audience}{%
\subsection{Audience}\label{audience}}

CoL ZooBank WoRMS? TW BHL users Taxonomists Librarians Citizen
Scientists Nomenclators Other Reseachers

(Results)

\hypertarget{examples-of-usages}{%
\subsubsection{Examples of usages}\label{examples-of-usages}}

\hypertarget{summary-of-scores}{%
\subsubsection{Summary of scores}\label{summary-of-scores}}

\hypertarget{benchmarks}{%
\subsubsection{Benchmarks}\label{benchmarks}}

\hypertarget{open-data-deliverables}{%
\subsubsection{Open data deliverables}\label{open-data-deliverables}}

(Discussion)

\hypertarget{conclusions}{%
\subsection{Conclusions}\label{conclusions}}

\begin{itemize}
\tightlist
\item
  Jigsaw puzzle where a name is the first piece.
\item
  We add more pieces
\end{itemize}

Advantages, impact

Missing parts

Ways to improve

How it helps for the other future developments

How to contribute?

Funding: skip it here? Same as above

\hypertarget{web-location-uris}{%
\section{Web location (URIs)}\label{web-location-uris}}

Homepage: bhlnames.globalnames.org

Wiki: github.com/gnames/bhlnames wiki (maybe, or nothing)

Download page: data or code? * code: https://github.com/gnames/bhlnames
* data: some place on Dryad?

Download mirror: n/a

Bug database: https://github.com/gnames/bhlnames/issues

Mailing list: https://gitter.io/gnames/something ?

Blog: http://globalnames.org

Vendor: N/A ask Matt

\hypertarget{technical-specification}{%
\section{Technical specification}\label{technical-specification}}

Platform: ?

Programming language: Go, JavaScript?

Operating system: GNU/Linux

Interface language: JavaScript?

Service endpoint: * https://bionames.globalnames.org/api/v1 *
https://apidoc.globalnames.org/bhlnames

\hypertarget{repository}{%
\section{Repository}\label{repository}}

Type: Git

Browse URI: https://github.com/gnames/bhlnames

Location: ?

Module: ?

Anonymous root: n/a

\hypertarget{usage-license}{%
\section{Usage license}\label{usage-license}}

MIT

\hypertarget{implementation}{%
\section{Implementation}\label{implementation}}

Implements specification: N/A

Audience:

\hypertarget{additional-information}{%
\section{Additional information}\label{additional-information}}

\hypertarget{acknowledgements}{%
\section{Acknowledgements}\label{acknowledgements}}

Thanks to you all. Rich Pyle for journal data

\hypertarget{author-contributions}{%
\section{Author contributions}\label{author-contributions}}

\hypertarget{references}{%
\section{References}\label{references}}

\hypertarget{figures}{%
\section{Figures}\label{figures}}

\hypertarget{tables}{%
\section{Tables}\label{tables}}

\hypertarget{supplementary-material}{%
\section{Supplementary material}\label{supplementary-material}}

\hypertarget{endnotes}{%
\section*{Endnotes}\label{endnotes}}
\addcontentsline{toc}{section}{Endnotes}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Bisby2000}{}}%
Bisby FA (2000) {The quiet revolution: Biodiversity informatics and the
Internet}. Science 289: 2309--2312.
\url{https://doi.org/10.1126/science.289.5488.2309}

\leavevmode\vadjust pre{\hypertarget{ref-Costantino2020}{}}%
Costantino G (2020) {BHL Improves the Speed and Accuracy of its
Taxonomic Name Finding Services with gnfinder}. Available from:
\url{https://blog.biodiversitylibrary.org/2020/07/bhl-improvestaxonomic-name-services-gnfinder.html}.

\leavevmode\vadjust pre{\hypertarget{ref-Gwinn2009}{}}%
Gwinn NE, Rinaldo C (2009) {The Biodiversity Heritage Library: Sharing
biodiversity literature with the world}. IFLA Journal 35: 25--34.
\url{https://doi.org/10.1177/0340035208102032}

\leavevmode\vadjust pre{\hypertarget{ref-Kalfatovic2019}{}}%
Kalfatovic MR, Costantino G, Rinaldo CA (2019) {The Biodiversity
Heritage Library: Unveiling a World of Knowledge About Life on Earth}.
In:, 352--355. \url{https://doi.org/10.1007/978-3-030-30760-8_32}

\leavevmode\vadjust pre{\hypertarget{ref-Page2010}{}}%
Page R (2010) {Finding scientific articles in a large digital archive:
BioStor and the Biodiversity Heritage Library}. Nature Precedings:
1--19. \url{https://doi.org/10.1038/npre.2010.4928.1}

\leavevmode\vadjust pre{\hypertarget{ref-Pilsk2010}{}}%
Pilsk SC, Person MA, Deveer JM, Furfey JF, Kalfatovic MR (2010) {The
Biodiversity Heritage Library: Advancing Metadata Practices in a
Collaborative Digital Library}. Journal of Library Metadata 10:
136--155. \url{https://doi.org/10.1080/19386389.2010.506400}

\leavevmode\vadjust pre{\hypertarget{ref-Richard2020}{}}%
Richard J (2020) {Improving Taxonomic Name Finding in the Biodiversity
Heritage Library}. Biodiversity Information Science and Standards 4.
\url{https://doi.org/10.3897/biss.4.58482}

\leavevmode\vadjust pre{\hypertarget{ref-Rinaldo2009}{}}%
Rinaldo C, Norton C (2009) {BHL, The Biodiversity Heritage Library: An
Expanding International Collaboration}. Nature Precedings.
\url{https://doi.org/10.1038/npre.2009.3620.1}

\leavevmode\vadjust pre{\hypertarget{ref-Smith2016}{}}%
Smith JE, Rinaldo CA (2016) {Collaborating on open science: The journey
of the Biodiversity Heritage Library}. Dobreva M, Schmidt B (Eds).
Information Services \& Use 35: 211--216.
\url{https://doi.org/10.3233/ISU-150785}

\end{CSLReferences}

\end{document}
